ipywidgets/widgets/trait_types/__init__.py

```diff
@@ -0,0 +1,185 @@
+_color_re = re.compile(r'#[a-fA-F0-9]{3}(?:[a-fA-F0-9]{3})?$')
+
+
+class Color(traitlets.Unicode):
+    """A string holding a valid HTML color such as 'blue', '#060482', '#A80'"""
+
+    info_text = 'a valid HTML color'
+    default_value = traitlets.Undefined
+
+    def validate(self, obj, value):
+        if value.lower() in _color_names or _color_re.match(value):
+            return value
+        self.error(obj, value)
+
+
+class Datetime(traitlets.TraitType):
+    """A trait type holding a Python datetime object"""
+
+    klass = dt.datetime
+    default_value = dt.datetime(1900, 1, 1)
+
+def datetime_to_json(pydt, manager):
+    """Serializes a Python datetime object to json.
+
+    Instantiating a JavaScript Date object with a string assumes that the
+    string is a UTC string, while instantiating it with constructor arguments
+    assumes that it's in local time:
+
+    >>> cdate = new Date('2015-05-12')
+    Mon May 11 2015 20:00:00 GMT-0400 (Eastern Daylight Time)
+    >>> cdate = new Date(2015, 4, 12) // Months are 0-based indices in JS
+    Tue May 12 2015 00:00:00 GMT-0400 (Eastern Daylight Time)
+
+    Attributes of this dictionary are to be passed to the JavaScript Date
+    constructor.
+    """
+    if pydt is None:
+        return None
+    else:
+        return dict(
+            year=pydt.year,
+            month=pydt.month - 1,  # Months are 0-based indices in JS
+            date=pydt.day,
+            hours=pydt.hour,       # Hours, Minutes, Seconds and Milliseconds are
+            minutes=pydt.minute,   # plural in JS
+            seconds=pydt.second,
+            milliseconds=pydt.microsecond / 1000
+        )
+
+def datetime_from_json(js, manager):
+    """Deserialize a Python datetime object from json"""
+    if js is None:
+        return None
+    else:
+        return dt.datetime(
+            js['year'],
+            js['month'] + 1,  # Months are 1-based in Python
+            js['date'],
+            js['hours'],
+            js['minutes'],
+            js['seconds'],
+            js['milliseconds'] * 1000
+        )
+
+datetime_serialization = {
+    'from_json': datetime_from_json,
+    'to_json': datetime_to_json
+}
+
+class InstanceDict(traitlets.Instance):
+    """An instance trait which coerces a dict to an instance.
+    
+    This lets the instance be specified as a dict, which is used to initialize the instance.
+    
+    Also, we default to a trivial instance, even if args and kwargs is not specified."""
+
+    def validate(self, obj, value):
+        if isinstance(value, dict):
+            return super(InstanceDict, self).validate(obj, self.klass(**value))
+        else:
+            return super(InstanceDict, self).validate(obj, value)
+
+    def make_dynamic_default(self):
+        return self.klass(*(self.default_args or ()),
+                          **(self.default_kwargs or {}))
+
+
+class EDict(eventful.Eventful, traitlets.Dict):
```

Comment:
This was just the naming convention used by @SylvainCorlay in https://github.com/ipython/traitlets/pull/278. I'm okay changing it though since the `Eventful`  prefix is more descriptive than `E`.